src/main/scala/nodes/learning/KernelRidgeRegression.scala

```diff
@@ -0,0 +1,343 @@
+package nodes.learning
+
+import workflow.LabelEstimator
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+import edu.berkeley.cs.amplab.mlmatrix.RowPartitionedMatrix
+import edu.berkeley.cs.amplab.mlmatrix.util.{Utils => MLMatrixUtils}
+
+import scala.util.Random
+
+class KernelRidgeRegression(
+  kernelBlockGenerator: KernelGenerator,
```

Comment:
@Vaishaal and I talked about this on the whiteboard briefly. I think I basically understand the math difference now. We can clarify on a whiteboard at some point. We should be a little careful with our use of the word "solves" since i think that's overloaded. Basically what we're saying is if A is a linear system with identical equations and unknowns (as in the case where A is the Kernel matrix, we want to find x which is an exact solution if one exists). We still use BCD to find this solution, rather than directly inverting A, but we are not minimizing the least squares loss.
In the case where A is not square (either A is overdetermined or underdetermined), we're going to find the X which minimizes the least-squares loss as you state. 

On the KernelGenerator bit, I think from a syntax perspective it's better to have this be an estimator. It also makes it easier to track usages of this RDD for auto-caching, etc. Also, a kernel function can be viewed as a degenerate kNN which is something I'd think of as an Estimator :). This gets the pipeline back to (roughly) how i wrote it before.